app/views/membership_applications/_reason_waiting.html.haml

```diff
@@ -2,32 +2,30 @@
   Do not make user press a "save" or "submit" button for these changes. (use AJAX)
 
 
-= form_for @membership_application, html: { id: 'member-app-waiting-reasons', multipart: true}, url: membership_application_path do |f|
-  .row
-    .container
-      .reason-waiting-information
-        = f.label( :member_app_waiting_reasons, t('membership_applications.need_info.reason_title'))
-        = f.collection_select(:member_app_waiting_reasons_id,
-                              AdminOnly::MemberAppWaitingReason.all,
-                              :id,
-                              reason_name_method,
-                              { include_blank: t('membership_applications.need_info.select_a_reason') },
-                              { class: 'reason-waiting-list',
-                                onchange: 'changed_reason()' },
-                              )
-
-  .row
-    .container
-      #other-text-field
-        = f.label( :reason_waiting_custom_text, t('membership_applications.need_info.other_reason_label') )
-        = f.text_field :custom_reason_text, onchange: 'changed_custom_text()'
+.row
+  .container
+    .reason-waiting-information
+      = label_tag :member_app_waiting_reasons, t('membership_applications.need_info.reason_title')
+      - collection = AdminOnly::MemberAppWaitingReason.all.to_a
+      - collection << AdminOnly::MemberAppWaitingReason.new(id: -1, "name_#{I18n.locale.to_s}": "#{@other_waiting_reason_text}")
+      = select_tag(:member_app_waiting_reasons,
+                   options_from_collection_for_select(collection, :id, reason_name_method,
+                                      @membership_application.member_app_waiting_reasons_id),
+                   { include_blank: t('membership_applications.need_info.select_a_reason'),
+                     class: 'reason-waiting-list' })
+
+.row
+  .container
+    #other-text-field
+      = label_tag :custom_reason_text, t('membership_applications.need_info.other_reason_label')
+      = text_field_tag :custom_reason_text, @membership_application.custom_reason_text
 
 
 :javascript
 
-  let reasons_list = document.getElementById('membership_application_member_app_waiting_reasons_id');  // the select HTML element (list)
+  let reasons_list = $('#member_app_waiting_reasons');  // the select HTML element (list)
```

Comment:
You're using both straight JSAPI as well as Jquery - thought it best to just use Jquery throughput.